.github/workflows/test-hosted-runners.yml

```
@@ -0,0 +1,52 @@
+name: Run Tests
+
+#
+# Create some files with script/new-artifact.sh and confirm they are properly packaged and uploaded
+# as artifacts with the actions.
+#
+# This is tested on all OS platforms where we have hosted runners.
+#
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  test:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Generate files
+      run: mkdir artifact && mkdir artifact2 && cd artifact && ../script/new-artifact.sh
+      shell: bash
+
+    - name: Upload artifact
```

Not a blocker but changing this name to more closely align with your Action's purpose might avoid some confusion with `actions/upload-artifact` 🤷🏻‍♂️

```suggestion
    - name: Upload Pages artifact
```